processors/src/main/java/io/opentelemetry/contrib/eventbridge/EventToSpanEventBridge.java

```diff
@@ -0,0 +1,130 @@
+ *       log.record.body}, as an escaped JSON string following the standard protobuf JSON encoding
+ *   <li>{@link LogRecordData#getTotalAttributeCount()} - {@link
+ *       LogRecordData#getAttributes()}.size() is mapped to span event attribute with key {@code
+ *       log.record.dropped_attributes_count}
+ * </ul>
+ */
+public final class EventToSpanEventBridge implements LogRecordProcessor {
+
+  private static final Logger LOGGER = Logger.getLogger(EventToSpanEventBridge.class.getName());
+
+  private static final AttributeKey<String> EVENT_NAME = AttributeKey.stringKey("event.name");
+  private static final AttributeKey<Long> LOG_RECORD_OBSERVED_TIME_UNIX_NANO =
+      AttributeKey.longKey("log.record.observed_time_unix_nano");
+  private static final AttributeKey<Long> LOG_RECORD_SEVERITY_NUMBER =
+      AttributeKey.longKey("log.record.severity_number");
+  private static final AttributeKey<String> LOG_RECORD_BODY =
+      AttributeKey.stringKey("log.record.body");
+  private static final AttributeKey<Long> LOG_RECORD_DROPPED_ATTRIBUTES_COUNT =
+      AttributeKey.longKey("log.record.dropped_attributes_count");
+
+  private EventToSpanEventBridge() {}
+
+  /** Create an instance. */
+  public static EventToSpanEventBridge create() {
+    return new EventToSpanEventBridge();
+  }
+
+  @Override
+  public void onEmit(Context context, ReadWriteLogRecord logRecord) {
+    LogRecordData logRecordData = logRecord.toLogRecordData();
+    String eventName = logRecordData.getAttributes().get(EVENT_NAME);
+    if (eventName == null) {
+      return;
+    }
+    if (!logRecordData.getSpanContext().isValid()) {
+      return;
+    }
+    Span currentSpan = Span.current();
+    if (!currentSpan.isRecording()) {
+      return;
+    }
+    currentSpan.addEvent(
+        eventName,
+        toSpanEventAttributes(logRecordData),
+        logRecordData.getTimestampEpochNanos(),
+        TimeUnit.NANOSECONDS);
+  }
+
+  private static Attributes toSpanEventAttributes(LogRecordData logRecord) {
+    AttributesBuilder builder =
+        logRecord.getAttributes().toBuilder().removeIf(key -> key.equals(EVENT_NAME));
+
+    builder.put(LOG_RECORD_OBSERVED_TIME_UNIX_NANO, logRecord.getObservedTimestampEpochNanos());
+
+    builder.put(LOG_RECORD_SEVERITY_NUMBER, logRecord.getSeverity().getSeverityNumber());
+
+    // Add bridging for logRecord.getSeverityText() if EventBuilder adds severity text setter
+
+    Value<?> body = logRecord.getBodyValue();
+    if (body != null) {
+      MarshalerWithSize marshaler = AnyValueMarshaler.create(body);
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      try {
+        marshaler.writeJsonTo(baos);
+      } catch (IOException e) {
+        LOGGER.log(Level.WARNING, "Error converting log record body to JSON", e);
+      }
+      builder.put(LOG_RECORD_BODY, new String(baos.toByteArray(), StandardCharsets.UTF_8));
```

Comment:
```suggestion
      builder.put(LOG_RECORD_BODY, baos.toString(StandardCharsets.UTF_8));
```

Reply:
Added in java 10. 🙁

Reply:
Oh yeah, I guess we need to use the String charset version then

https://docs.oracle.com/javase/8/docs/api/?java/io/ByteArrayOutputStream.html